### PR TITLE
other(find-all): throw on empty attributes

### LINKED ISSUE #11858: Sqlite: using `attributes: []` produces broken select statement
## Issue Description

The following code will generate an invalid sql statement when running against an sqlite database

```js
const found = await Foo.findOne({
  attributes: [],
})
```

The generated sql is something like: 

```sql
SELECT  FROM "database"."table"
```

A common workaround is adding a fixed result column, like `SELECT 1 FROM ...`, although I'm not sure how that will play with the next step, when the records are parsed.

### Is this issue dialect-specific?

- [ ] No. This issue is relevant to Sequelize as a whole.
- [x] Yes. This issue only applies to the following dialect(s): SQLITE
- [ ] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [x] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

### PR DESCRIPTION
**Pull request by papb**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [X] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [X] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [X] Have you added new tests to prevent regressions?
- [X] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Closes #11858

This provides a much better error message for the situation described in #11858.

However it is technically a breaking change because on Postgres running `SELECT FROM "Foos"` does not throw (simply yielding an empty array). However from the Sequelize point of view, it should throw (IMO), since why would you run a SELECT query and fetch nothing?

*Note:* The commit as in PR title `other(find-all): throw on empty attributes` is not allowed by commitlint because `other` is not a valid scope type. However, I am unsure of what choice to make among the valid options, which are *build, ci, docs, feat, fix, perf, refactor, revert, style and test*, so I leave it for you to decide @sushantdhiman. I ended up putting `fix(find-all)` in the actual commit message, but on squash-and-merge that won't matter much.

### CODE REVIEW

**Comment by sushantdhiman** [comment]
Tests are breaking. I don't think it is a good idea, what if you got includes or scopes are injecting attributes.

**Comment by papb** [comment]
Hi @sushantdhiman, you are right, however I still think something could be done about these errors, so I worked on another approach. Tests are passing now, please review again, thanks!

File: lib/dialects/abstract/query-generator.js [hunk 1]
```diff
@@ -2019,7 +2020,26 @@ class QueryGenerator {
     return { mainQueryOrder, subQueryOrder };
   }
 
+  _ensureNonemptySelectClause(attributes, extraInfo = {}) {
```
Comment (sushantdhiman) [inline lib/dialects/abstract/query-generator.js]:
```suggestion
_throwOnEmptyAttributes(attributes, extraInfo = {}) {
  if (attributes.length > 0) return;

  const asPart = extraInfo.as && `as ${extraInfo.as}` || '';
  const namePart = extraInfo.modelName && `for model '${extraInfo.modelName}'` || '';
  const message = `Attempted a SELECT query ${namePart} ${asPart} without selecting any columns`;

  throw new sequelizeError.QueryError(message.replace(/  +/g, ' '));
}
```

Comment (papb) [reply lib/dialects/abstract/query-generator.js]: @sushantdhiman Done, thanks :+1:

File: docs/manual/other-topics/upgrade-to-v6.md [hunk 2]
```diff
@@ -46,6 +46,21 @@ This method now tests for equality with [`_.isEqual`](https://lodash.com/docs/4.
   await instance.save(); // will save
 ```
 
+#### Finder methods with empty attributes
```
Comment (sushantdhiman) [inline docs/manual/other-topics/upgrade-to-v6.md]: This should be treated as a bug fix, no need to mention this here. Apart from postgres it was not working, something like a type error.

When next release is pushed I will update the changelog along with other commits.

Comment (papb) [reply docs/manual/other-topics/upgrade-to-v6.md]: Hmmm, if you say so :+1:

**Pull request merged**